features/armv8_stack_overflow_guards/README.md

```diff
@@ -0,0 +1,194 @@
+# ARMV8 STACK OVERFLOW GUARDS
+This example illustrates how to use the armv8 MSP and PSP limit registers to use hardware to protect against stack overflows.
+
+Historically, stack overflows have plagued the embedded development process and have wasted engineering hours trying to root causing the stack overflow itself. Software solutions have been implemented to detect these conditions, but still fall short of detecting all stack overflows, under all conditions.
+
+FreeRTOS utilizes the 'magic number' approach.  They write a signature to each tasks stack location, and then check the last word in the stack, to see if it is been over-written. They also implement a high watermark reading, where they scan the stack to see the max depth at any given time.   There are several downfalls of this method:
```

Comment:
Suggest adding a reference to the FreeRTOS docs: https://www.freertos.org/Stacks-and-stack-overflow-checking.html
Would also highlight their overflow checking is not recommended for production: "Stack overflow checking introduces a context switch overhead so its use is only recommended during the development or testing phases. "